cmd/restic/cmd_forget.go

```diff
@@ -213,15 +191,21 @@ func runForget(opts ForgetOptions, gopts GlobalOptions, args []string) error {
 		}
 	}
 
-	if removeSnapshots > 0 && opts.Prune {
-		if !gopts.JSON {
-			Verbosef("%d snapshots have been removed, running prune\n", removeSnapshots)
+	if len(removeSnIDs) > 0 {
+		if !opts.DryRun {
+			err := DeleteFiles(gopts, false, repo, removeSnIDs, restic.SnapshotFile)
+			if err != nil {
+				return err
+			}
+		} else {
+			if !gopts.JSON {
+				Printf("Would have removed the following snapshots:\n%v\n\n", removeSnIDs)
+			}
 		}
 		if !opts.DryRun {
```

Comment:
Prune should only be run when `opts.Prune` is set.

Reply:
Ouch - sorry, missed that when I extracted this from the prune reimplementation..
Thanks for finding this! Is corrected now.